### PR TITLE
Fixes #10171: If the Rudder Web Interface is interrupted during the second step of reports archiving, then it won't never succeed in archiving again

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/10171

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala [hunk 1]
```diff
@@ -235,30 +235,84 @@ class ReportsJdbcRepository(jdbcTemplate : JdbcTemplate) extends ReportsReposito
...
+    jdbcTemplate.query("select id from ruddersysevents order by id asc limit 1",
+          IdMapper).asScala match {
+      case seq if seq.size > 1 => Failure("Too many answer for the lowest report in the database")
+      case seq => Full(seq.headOption)
+    }
+  }
+
+  private[this] def getHighestIdBeforeDate(date : DateTime) : Box[Option[Long]] = {
+    println("getHighestIdBeforeDate")
+    val query = s"select id from RudderSysEvents where executionTimeStamp < '${date.toString("yyyy-MM-dd")}' order by id desc limit 1"
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala]: RudderSysEvents should be lower case

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**